mrp_partial_production/wizard/mrp_product_produce.py

```diff
@@ -1,86 +1,28 @@
 # -*- coding: utf-8 -*-
-##############################################################################
-#
-#    OpenERP, Open Source Management Solution
-#    Copyright (C) 2004-2010 Tiny SPRL (<http://tiny.be>).
-#
-#    This program is free software: you can redistribute it and/or modify
-#    it under the terms of the GNU Affero General Public License as
-#    published by the Free Software Foundation, either version 3 of the
-#    License, or (at your option) any later version.
-#
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# Copyright 2017 Vauxoo (https://www.vauxoo.com) <info@vauxoo.com>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
 
-from openerp import models, api, fields, _
-from openerp.exceptions import Warning as UserError
-import openerp.addons.decimal_precision as dp
+from odoo import api, models, _
+from odoo.exceptions import Warning as UserError
 
 
-class MrpProductProduct(models.TransientModel):
+class MrpProductProduce(models.TransientModel):
     _inherit = "mrp.product.produce"
 
-    @api.multi
-    def on_change_qty(self, product_qty, consume_lines):
-        """When changing the quantity of products to be produced it will
-        recalculate the number of raw materials needed according
-        to the scheduled products and the already consumed/produced products
-        It will return the consume lines needed for the products to be produced
-        which the user can still adapt
-        """
-        res = super(MrpProductProduct, self).on_change_qty(product_qty,
-                                                           consume_lines)
-        prod_obj = self.env["mrp.production"]
-        production = prod_obj.browse(self._context['active_id'])
-        if production.qty_available_to_produce < product_qty:
-            raise UserError(_('''You cannot produce more than available to
-                              produce for this order '''))
-
+    @api.model
+    def default_get(self, fields):
+        res = super(MrpProductProduce, self).default_get(fields)
+        if self._context and self._context.get('active_id'):
```

Comment:
with new api you do not need validate self._context or in what case is not self._context?

Reply:
it is not necessary validate self._context. I remove it